tests/integration/integration_suite_test.go

```diff
@@ -1,13 +1,187 @@
-package integration_test
+package integration
```

Comment:
I think we still haven't decided if we were dealing with e2e or integration tests here 😅 
I'm checking a lot of intermediate steps and not with the all stack loaded each time, so maybe that doesn't qualify as e2e.

Reply:
Right, that is fine, I was thinking we could build out a folder structure with placeholders so developers know to think about a distinction.

Reply:
@adejanovski @jsanda I wanted to add this note since it seems relevant.  We should consider adding use of something like this [fake-client](https://github.com/kubernetes/client-go/blob/master/examples/fake-client/main_test.go) in our integration testing to alleviate having all resources available every time.  This will greatly speed up our local runs where we aren't performing a specific verification with another resource that can be mocked.

When we discuss `e2e`, I see all the resources running and we inject known inputs (i.e. state changes, configs, etc.) expecting known outputs (state, configs, etc.).  With `integration` tests, specific components are involved and where needed, the others can be `mocked`.  If we take this approach early on, it will make things much more manageable for local integration testing.